changelogs/unreleased/gh-tarantool-10142-box-ctl-wal-sync-lua.md

```diff
@@ -0,0 +1,7 @@
+## feature/lua
+
+* Added `box.ctl.wal_sync()` function which waits until
+  all submitted writes are successfully flushed to the disk.
+  Throws an error if write fails. The main goal is to
+  provide a way to synchronize vclock between instances
+  primarily for switching masters (gh-10142).
```

Comment:
IMO the changelog and doc request are misleading.

In doc request:
> It is primarily needed for synchronisation between
old and new masters or actually any other situations when user
need consistent vclock between instances.

In changelog:
> The main goal is to
  provide a way to synchronize vclock between instances
  primarily for switching masters 

Having read these, I'd believe that `wal_sync` somehow does a synchronization of vclocks between instances.

Let's better rephrase these places in both doc request and changelog. For example, `After the function is executed one may reliably use box.info.vclock for comparisons when choosing a new master`.

Reply:
Also the changelog should go to `feature/core`, as it doesn't have anything to do with lua itself.
(`core` and not `box` since we've agreed to merge these two paragraphs. For example, see the latest release changelogs https://github.com/tarantool/tarantool/releases)

Reply:
Sure. I have no objections to the proposed fix so I added it. I hope it's more clear now what the function does. Fixed.